week2/ex1_done.py

```diff
@@ -0,0 +1,39 @@
+# 1.	Write a function that will parse a url query string like "page=2&foo=bar&x=y&utm_source=ss&utm_content=ccc" and return a query string where all the params are sorted by their names like this "foo=bar&page=2&x=y" and all the "utm_" parameters are removed.
+
+
+def parse_url(url):
+    string_splitted = url.split("&")
+    params_dict = {}
+    for entry in string_splitted:
```

Comment:
What you are doing here is basically filtering a list. This is very common operation in Python and it is done best with [list comprehensions](https://docs.python.org/3.6/tutorial/datastructures.html#list-comprehensions):

```python
filtered = [x for x in string_splitted if not x.startswith("utm_")]
```

Another approach is to use [`filter()`](http://book.pythontips.com/en/latest/map_filter.html):
```python
filtered = filter(lambda x: not x.startswith("utm_"), string_splitted)
```


The basic operations on lists are: `filter()`, `map()`, `reduce()` ([read](http://book.pythontips.com/en/latest/map_filter.html)). Keep them in mind when dealing with lists. You can solve almost any task just with this three functions.